source/Optimization.rst

```diff
@@ -0,0 +1,37 @@
+Optimisation
+===============
+
+
+PerTickCache
+----------------
+This function is used to retrieve values from the cache if some value is already computed. If the value is not present in the cache, it computes the value using a user-defined function and stores it in the cache.
+This is extremely useful in reducing the simulation time when we repeatedly have to query the context to compute the value for every agent in a tick.
+
+For example, consider the case in the simple SIR model where we have to find the fraction of infected individuals around an agent in a particular place. Here the infected fraction of all agents in the same place would be the same. So while running the simulation, we have to compute the same value repeatedly for all agents in that place. We need to query the context for all agents during each tick, thereby increasing the simulation time and computational resource required.
+Here when we implement PerTickCache, the infected fraction will be computed for all places as and when needed during each tick, and that value will be stored in the cache to be used for all the agents in that place.
+
+If the value is not computed for a place, the PerTickCache function will compute the value using a user-defined function. 
+
+Example for PerTickCache implementation:
```

Comment:
Very good idea. However, you are writing in markdown! You don't need to put in a chunk of code with comments inside it. Please reformat this so that you first describe what a function is doing in markdown, and then just include the code for that function below it. Do these for both the comments below: basically, be a little more verbose. After you are done, explain again that the ``computeTheValue`` function is only called **once** per tick per location, and how that can speed things up when you have a lot of agents in a location.